main.py

```diff
@@ -74,39 +97,67 @@ def contact_list():
     contacts = session.query(Contact).all()
     return contacts
 
-def get_contact_by_name(name):
-    global contacts
-    if not (name in contacts):
-        return None
-    return contacts[name]
+def get_contact_by_name():
+    if 'CONTACT_NAME' in state:
+        contacts = session.query(Contact).filter_by(name = state['CONTACT_NAME']).all()
+        if contacts:
+            for contact in contacts:
+                print(contact.name, contact.number)
```

Comment:
We need to improve the usability of this.
Please collect Ideas so we can discuss.